cli_tools/gce_ovf_import/ovf_utils/ovf_utils.go

```diff
@@ -250,31 +377,66 @@ func GetOVFDescriptorAndDiskPaths(ovfDescriptorLoader domain.OvfDescriptorLoader
 func GetOSId(ovfDescriptor *ovf.Envelope) (string, error) {
 
 	if ovfDescriptor.VirtualSystem == nil {
-		return "", daisy.Errf("VirtualSystem must be defined to retrieve OS info")
+		return "", daisy.Errf("OVF descriptor error: VirtualSystem must be defined to retrieve OS info. Use --os flag to specify OS")
 	}
 	if ovfDescriptor.VirtualSystem.OperatingSystem == nil ||
 		len(ovfDescriptor.VirtualSystem.OperatingSystem) == 0 {
-		return "", daisy.Errf("OperatingSystemSection must be defined to retrieve OS info")
-	}
-	if ovfDescriptor.VirtualSystem.OperatingSystem[0].OSType == nil {
-		return "", daisy.Errf("OperatingSystemSection.OSType must be defined to retrieve OS info")
-	}
-	var osID string
-	var validOSType bool
-	osType := *ovfDescriptor.VirtualSystem.OperatingSystem[0].OSType
-	if osID, validOSType = ovfOSTypeToOSID[osType]; !validOSType {
-		if osIDCandidates, hasOSIDCandidates := noMappingOSTypes[osType]; hasOSIDCandidates {
-			return "",
-				daisy.Errf(
-					"cannot determine OS from osType attribute value `%v` found in OVF descriptor. Use --os flag to specify OS for this VM. Potential valid values for given osType attribute are: %v",
-					osType,
-					strings.Join(osIDCandidates, ", "),
-				)
+		return "", daisy.Errf("OVF descriptor error: OperatingSystemSection must be defined to retrieve OS info. Use --os flag to specify OS")
+	}
+	if ovfDescriptor.VirtualSystem.OperatingSystem[0].OSType == nil && ovfDescriptor.VirtualSystem.OperatingSystem[0].ID == 0 {
+		return "", daisy.Errf("OVF descriptor error: OperatingSystemSection.OSType or OperatingSystemSection.ID must be defined to retrieve OS info. Use --os flag to specify OS")
+	}
+
+	var osInfoFromOSType, osInfoFromOSID *osInfo
+	if ovfDescriptor.VirtualSystem.OperatingSystem[0].OSType != nil && *ovfDescriptor.VirtualSystem.OperatingSystem[0].OSType != "" {
+		if osInfoFromOSTypeValue, ok := ovfOSTypeToOSID[*ovfDescriptor.VirtualSystem.OperatingSystem[0].OSType]; ok {
+			osInfoFromOSType = &osInfoFromOSTypeValue
+		}
+	}
+
+	if ovfDescriptor.VirtualSystem.OperatingSystem[0].ID != 0 {
+		if osInfoFromOSIDValue, ok := ovfOSIDToImporterOSID[ovfDescriptor.VirtualSystem.OperatingSystem[0].ID]; ok {
+			osInfoFromOSID = &osInfoFromOSIDValue
 		}
-		return "", daisy.Errf("osType attribute value `%v` found in OVF descriptor cannot be mapped to an OS supported by Google Compute Engine. Use --os flag to specify OS for this VM", osType)
 	}
 
-	return osID, nil
+	if osInfoFromOSType == nil && osInfoFromOSID == nil {
+		return "", daisy.Errf("cannot determine OS from OVF descriptor. Use --os flag to specify OS")
+	}
+
+	osInfo := getMoreSpecificOSInfo(osInfoFromOSType, osInfoFromOSID)
+
+	if !osInfo.hasImporterOSIDs() {
```

Comment:
What about the "Other" OS return value? Shouldn't that mean that we haven't detected it?

Reply:
I think "Other" differs from "Unknown" (ID=0, not included). I'm not sure there's much value in keeping "Other" except that OVF descriptor is telling us there is an OS there but it's not listed. "Unknown" is basically `null` value and is thus not included.

I don't have strong preference on whether to keep "Other/Other 64-bit" or not.

Reply:
I'd probably omit `Other`, since it would result in an error message of `cannot determine OS from OVF descriptor.`, which seems more helpful than `operating system "Other" detected but is not supported by...`

Reply:
Removed "Other" and "Other 64-bit"